soccer/src/soccer/planning/planner/collect_path_planner.cpp

```diff
@@ -243,6 +266,246 @@ Trajectory CollectPathPlanner::coarse_approach(
     return coarse_path;
 }
 
+void CollectPathPlanner::calc_delta_pos_for_dir(BallState ball, RobotInstant start_instant,
+                                                rj_geometry::Point* delta_robot_pos,
+                                                rj_geometry::Point* face_pos) {
+    // Get angle between target and normal hit
+    Point normal_face_vector = ball.position - start_instant.position();
+
+    // Since we loose the sign for the angle between call, there are two
+    // possibilities
+
+    *delta_robot_pos = Point(0, 0);
+    *face_pos = start_instant.position() + Point::direction(normal_face_vector.angle()) * 10;
+}
+
+Trajectory CollectPathPlanner::intercept(const PlanRequest& plan_request,
+                                         RobotInstant start_instant,
+                                         const rj_geometry::ShapeSet& static_obstacles,
+                                         const std::vector<DynamicObstacle>& dynamic_obstacles,
+                                         rj_geometry::Point delta_pos,
+                                         rj_geometry::Point face_pos) {
+    const double max_ball_angle_change_for_path_reset =
+        settle::PARAM_max_ball_angle_for_reset * M_PI / 180.0f;
+
+    BallState ball = plan_request.world_state->ball;
+
+    // If the ball changed directions or magnitude really quickly, do a reset of
+    // target
+    if (average_ball_vel_.angle_between(ball.velocity) > max_ball_angle_change_for_path_reset ||
+        (average_ball_vel_ - ball.velocity).mag() > settle::PARAM_max_ball_vel_for_path_reset) {
+        first_intercept_target_found_ = false;
+        average_ball_vel_initialized_ = false;
+    }
+
+    // Try find best point to intercept using brute force method
+    // where we check ever X distance along the ball velocity vector
+    //
+    // Disallow points outside the field
+    const Rect& field_rect = FieldDimensions::current_dimensions.field_rect();
+
+    std::optional<Point> ball_intercept_maybe;
+    RJ::Seconds best_buffer = RJ::Seconds(-1.0);
+
+    int num_iterations =
+        std::ceil((settle::PARAM_search_end_dist - settle::PARAM_search_start_dist) /
+                  settle::PARAM_search_inc_dist);
+
+    for (int iteration = 0; iteration < num_iterations; iteration++) {
```

Comment:
instead of calculating number of iterations can we do all the "math" in the for loop? Like start at start, end at end, and increment by increment for each part of for loop, respectively?